### PR TITLE
[quant][improvement] Check the fixedqparam op qconfig based on backend_config

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #87425

Summary:
Previously we hardcoded the supported observers for fixedqparam ops, this PR changes that to take the information from BackendConfig,
this allows users to customize the support for fixed qparam ops

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_change_backend_config_for_fixed_qparam_ops

Reviewers:

Subscribers:

Tasks:

Tags:

unlinked from diff since it's too hard to land

### CODE REVIEW

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40573882).

**Review by andrewor14 — APPROVED** [review]

File: test/quantization/fx/test_quantize_fx.py [hunk 1]
```diff
@@ -5223,6 +5223,31 @@ def forward(self, x):
...
+                self.tanh = torch.nn.Tanh()
+
+            def forward(self, x: torch.Tensor):
+                x = self.tanh(x)
+                return x
+
+        model = M().eval()
+        # we set a global default_qconfig, which will be ignored since the backend
+        # we defined doesn't support anything
+        qconfig_mapping = QConfigMapping().set_global(default_qconfig)
```
[the remark this answers is not in the document]
Comment (jerryzh168) [reply test/quantization/fx/test_quantize_fx.py:5241]: Yes

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40573882).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40573882).

**Comment by jerryzh168** [comment]
@pytorch merge -f "failing test doesn't seem related"

**Pull request closed without merging**